jbuild-workspace.dev

```diff
@@ -1,5 +0,0 @@
-;; This file is used by `make all-supported-ocaml-versions`
```

Comment:
@rgrinberg We had to remove this because of [this problem](https://github.com/janestreet/jbuilder/issues/318) with `jbuilder`. Having a workspace.dev file makes it impossible to build sandboxed dependencies in a local cache if those dependencies build with jbuilder because they will climb the file system searching for a `jbuild-workspace(.*)` file.